pkg/cache/tas_flavor_snapshot.go

```diff
@@ -560,35 +591,61 @@ func (s *TASFlavorSnapshot) findTopologyAssignment(
 	} else {
 		selector = labels.Everything()
 	}
-	// phase 1 - determine the number of pods which can fit in each topology domain
+	// phase 1 - determine the number of pods and chunks which can fit in each topology domain
 	s.fillInCounts(
 		requests,
 		assumedUsage,
+		chunkSize,
+		chunkLevelIdx,
 		simulateEmpty,
 		append(podSetTolerations, s.tolerations...),
 		selector,
 		requiredReplacementDomain,
 	)
 
 	// phase 2a: determine the level at which the assignment is done along with
-	// the domains which can accommodate all pods
-	fitLevelIdx, currFitDomain, reason := s.findLevelWithFitDomains(levelIdx, required, count, unconstrained)
+	// the domains which can accommodate all pods/chunks
+	fitLevelIdx, currFitDomain, reason := s.findLevelWithFitDomains(levelIdx, required, count, chunkSize, unconstrained)
 	if len(reason) > 0 {
 		return nil, reason
 	}
 
 	// phase 2b: traverse the tree down level-by-level optimizing the number of
 	// topology domains at each level
 	// if unconstrained is set, we'll only do it once
-	currFitDomain = s.updateCountsToMinimum(currFitDomain, count, unconstrained)
+	currFitDomain = s.updateChunkCountsToMinimum(currFitDomain, count, chunkSize, unconstrained)
 	for levelIdx := fitLevelIdx; levelIdx+1 < len(s.domainsPerLevel); levelIdx++ {
-		lowerFitDomains := s.lowerLevelDomains(currFitDomain)
-		sortedLowerDomains := s.sortedDomains(lowerFitDomains, unconstrained)
-		currFitDomain = s.updateCountsToMinimum(sortedLowerDomains, count, unconstrained)
+		if levelIdx < chunkLevelIdx {
+			lowerFitDomains := s.lowerLevelDomains(currFitDomain)
+			sortedLowerDomains := s.sortedDomains(lowerFitDomains, unconstrained)
+			currFitDomain = s.updateChunkCountsToMinimum(sortedLowerDomains, count, chunkSize, unconstrained)
+		} else {
+			newCurrFitDomain := make([]*domain, 0)
+			for _, domain := range currFitDomain {
+				lowerFitDomains := domain.children
+				sortedLowerDomains := s.sortedDomains(lowerFitDomains, unconstrained)
+
+				addCurrFitDomain := s.updateCountsToMinimum(sortedLowerDomains, domain.state, unconstrained)
+				newCurrFitDomain = append(newCurrFitDomain, addCurrFitDomain...)
+			}
+			currFitDomain = newCurrFitDomain
+		}
 	}
 	return s.buildAssignment(currFitDomain), ""
 }
 
+func getChunkSizeWithDefault(podSetTopologyRequest *kueue.PodSetTopologyRequest, defaultChunkSize int32) (int32, string) {
```

Comment:
Always called with same default. Can we get rid of this, and just document in the function?[

```suggestion
func getChunkSizeWithDefault(podSetTopologyRequest *kueue.PodSetTopologyRequest) (int32, string) {
```